### ISSUE TITLE
Appium 1.6.4@beta exits with Original error: Command 'java -version' exited with code 2

### ISSUE THREAD

**Issue by Brad19** [appium/java-client#661]
I'm trying to run my appium tests for android in emulator using appium 1.6.4@beta. The issue is when I run my code it stops running and exits with the message

> org.openqa.selenium.WebDriverException:
An unknown server-side error occurred while processing the command. Original error: Command 'java -version' exited with code 2 (WARNING: The server did not provide any stacktrace information)

I'm not sure what went wrong. Is this the issue with Appium or Jdk. 
I've tried the following
1. Commenting the capability for "automationNation" and ran that.
2. Changed the JAVA_HOME from /Library/Java/JavaVirtualMachines/jdk1.8.0_131.jdk/Contents/Home/bin rather than just Home.
3. Closed intelliJ opened it
4. I've verified the sdk settings from Project structure and it points to the right jdk home folder.
5. I tried upgrading appium to 1.6.5 

Is there any issue with Jdk1.8_0_131 with macSierra? Should I need to downgrade it. Do I miss any installations. I've installed based on this (http://chase-seibert.github.io/blog/2017/01/06/appium-react-native-quickstart.html)?

However my code works perfectly fine for IOS. I've verified the sdk settings from Project structure and it points to the right jdk home folder. I've spent nearly 2 days for this. If anyone can help me, that'll be really great. 

## Environment

Jdk: Jdk- 1.8.0_131
Env variables : 
export ANDROID_HOME=/Users/thiruba/Library/Android/sdk
export PATH=${PATH}:${ANDROID_HOME}/tools
export PATH=${PATH}:${ANDROID_HOME}/platform-tools
export JAVA_HOME=$(/usr/libexec/java_home)
Appium server: 1.6.4beta
capability settings:
capabilities.setCapability("deviceName", "emulator-5554");
                        capabilities.setCapability("platformName", "Android" );
                        capabilities.setCapability("platformVersion", "7.1.1");
                        capabilities.setCapability("appForAndroid", appForAndroid.getAbsolutePath());
                        capabilities.setCapability("appPackage", "xx.xxx.xxxxx");
                        capabilities.setCapability("appActivity","xx.xx.xxxx.MainActivity");
                        capabilities.setCapability("androidDeviceReadyTimeout", 30);
                        capabilities.setCapability("deviceReadyTimeout", 5);
                        capabilities.setCapability("automationNation", "uiautomator2");
OS: macOs sierra 10.12.5
XCode: 8.3.3
node version: v8.1.2
Android emulator: API 25 and Android 7.1.1
Project type: Maven
Pom.xml:
`<dependency>
            <groupId>org.seleniumhq.selenium</groupId>
            <artifactId>selenium-java</artifactId>
            <version>2.53.0</version>
            <scope>Test</scope>
        </dependency>
        <dependency>
            <groupId>org.testng</groupId>
            <artifactId>testng</artifactId>
            <version>6.10</version>
            <scope>Test</scope>
        </dependency>
        <dependency>
            <groupId>io.appium</groupId>
            <artifactId>java-client</artifactId>
            <version>2.1.0</version>
        </dependency>
`
## Exception stacktraces
`objc[11277]: Class JavaLaunchHelper is implemented in both /Library/Java/JavaVirtualMachines/jdk1.8.0_131.jdk/Contents/Home/bin/java (0x10edf54c0) and /Library/Java/JavaVirtualMachines/jdk1.8.0_131.jdk/Contents/Home/jre/lib/libinstrument.dylib (0x10eebd4e0). One of the two will be used. Which one is undefined.
[TestNG] Running:
  /Users/xxxx/Library/Caches/IntelliJIdea2017.1/temp-testng-customsuite.xml

org.openqa.selenium.WebDriverException: An unknown server-side error occurred while processing the command. Original error: Command 'java -version' exited with code 2 (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 204 milliseconds
Build info: version: '2.53.0', revision: '35ae25b1534ae328c771e0856c93e187490ca824', time: '2016-03-15 10:43:46'
System info: host: 'SYDMAC33.gateway', ip: '10.0.0.37', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.12.5', java.version: '1.8.0_131'
Driver info: Appium$1

	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:206)
	at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:158)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:678)
	at io.appium.java_client.AppiumDriver.execute(AppiumDriver.java:180)
	at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:249)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:131)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:158)
	at io.appium.java_client.AppiumDriver.<init>(AppiumDriver.java:109)
	at Appium$1.<init>(Appium.java:57)
	at Appium.setUp(Appium.java:57)
	at TestAppium.initialize(TestAppium.java:30)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:515)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:217)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:144)
	at org.testng.internal.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:169)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:756)
	at org.testng.TestRunner.run(TestRunner.java:610)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:387)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:382)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1293)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1218)
	at org.testng.TestNG.runSuites(TestNG.java:1133)
	at org.testng.TestNG.run(TestNG.java:1104)
	at org.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:72)
	at org.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:127)
`

## Link to Appium logs

Here are the logs from Appium server
....
`[debug] [AndroidDriver] AndroidDriver version: 1.23.0
[BaseDriver] The following capabilities were provided, but are not recognized by appium: appForAndroid, automationNation.
[BaseDriver] Session created with session id: 44270354-8a17-46a9-8347-7da53d848460
[debug] [AndroidDriver] Getting Java version
[debug] [AndroidDriver] Shutting down Android driver
[debug] [AndroidDriver] Called deleteSession but bootstrap wasn't active
[MJSONWP] Encountered internal error running command: Error: Command 'java -version' exited with code 2
    at ChildProcess.<anonymous> (../../lib/teen_process.js:82:19)
    at emitTwo (events.js:125:13)
    at ChildProcess.emit (events.js:213:7)
    at maybeClose (internal/child_process.js:897:16)
    at Socket.stream.socket.on (internal/child_process.js:340:11)
    at emitOne (events.js:115:13)
    at Socket.emit (events.js:210:7)
    at Pipe._handle.close [as _onclose] (net.js:548:12)
`

**Comment by SrinivasanTarget** [comment]
@Brad19 change caps from ```automationNation``` to ```automationName``` and give a try with latest version of appium server 1.6.6beta and client 5.0.0Beta9.

**Comment by Brad19** [comment]
oops. My bad. Sorry!!!. I think it should be typo. But tried with automationName and it didn't work. I finally downgraded to Jdk1.8.0_121 and it launched the app successfully which is good, but the test is failed due to 
`org.openqa.selenium.WebDriverException: An unknown server-side error occurred while processing the command. Original error: Command '/Users/xxxx/Library/Android/sdk/build-tools/23.0.1/aapt dump badging /usr/local/lib/node_modules/appium/node_modules/appium-uiautomator2-driver/uiautomator2/appium-uiautomator2-server-v0.1.5.apk' exited with code 1 (WARNING: The server did not provide any stacktrace information)`.  But IOS is fine. I found that someone is having similar issue (https://github.com/appium/appium/issues/8641). But I didn't find the solution.
